README.md

```diff
@@ -1,64 +1,199 @@
-# CrateDB + Apache Spark
+# Machine Learning with CrateDB and Apache Spark
 
-CrateDB is a distributed SQL database system which is capable of managing huge amounts of data in real time.
-CrateDB provides an easy, horizontal scalable and system independent solution built for IoT applications.
-This project explains the necessary steps to combine CrateDB with Apache Spark, a lightning fast cluster system which offers a variety operations for BigData.
-A really poplar feature of Spark is machine learning, which will be covered here.
-This project explains the necessary steps to apply machine learning algorithms on big data.
-For storage and ingestion, CrateDB is the software to store and query an incredible amount of data in real time.
-For machine learning, Apache Spark is the quick and easy way to produce machine learning models.
+## About
 
+[CrateDB](http://crate.io) is a distributed SQL database system which is capable of managing huge amounts of data in real time.
+It is an easy, horizontal scalable, efficient and system independent solution built for IoT applications.
+
+Though CrateDB is a pretty solid choice for managing BigData, it was not built for performing general data analysis.
+This project adds this feature to CrateDB by creating a connection with Apache Spark as data processing tool.
+
+[Apache Spark](http://spark.apache.org) is a cluster computing engine which offers lots and lots of various data operations.
+All data operations are done in a distributed way, where all computations are done in RAM, which performs about 100x times faster than the previous Apache Hadoop system.
+A really poplar feature of Spark is machine learning, which will mainly be covered here.
 
 Since the theory may be complex at first sight, a simple use case helps to explain the theory behind.
 
-## Requirements
+### Known Issues and Limits
```

Comment:
I would write a short section about the use case you want to implement before talking about the limitations, because some of the limitations assume knowledge about the use case.